tests/unit/index-nodetest.js

```diff
@@ -42,11 +50,41 @@ describe('the index', function() {
               type: 'file-hash',
               filePattern: 'eeee'
             }
+          },
+          commandLineArgs: {}
+        }
+      };
+
+      return assert.isFulfilled(plugin.configure.call(plugin, context));
+    });
+
+    it('sets the revision if it exists in the command line args', function() {
+      var plugin = subject.createDeployPlugin({
+        name: 'revision-key'
+      });
+
+      var context = {
+        deployment: {
+          ui: {
+            write: function() {},
+            writeLine: function() {}
+          },
+          config: {
+            'revision-key': {
+              type: 'file-hash',
+              filePattern: 'eeee'
+            }
+          },
+          commandLineArgs: {
+            revision: 'some-revision'
           }
         }
       };
 
       return assert.isFulfilled(plugin.configure.call(plugin, context))
+        .then(function(result) {
+          assert.equal(result.revision, 'some-revision');
```

Comment:
Ah, I understand the intent now. This is not how I envisioned this. The configure hook should be about validating config and applying defaults. In this case, the default being applied comes from the command line args.

The return values that are merged into context should be reserved for new data being produced by the pipeline.

So, I think the issue here is that we don't actually want to configure the revision key plug-in with the command line args, we want to configure the thing that will use the revision key, the redis plugin. So the configure hook of `ember-cli-deploy-redis` would apply a default revisionKey config function; something like:

```js
function(context){
  return context.deployment.revisionKey || context.deployment.commandLineArgs["revision"];
}
```